Mathlib/Data/Polynomial/EraseLead.lean

```diff
@@ -245,6 +245,29 @@ theorem induction_with_natDegree_le (P : R[X] → Prop) (N : ℕ) (P_0 : P 0)
       exact Nat.succ_ne_zero _
 #align polynomial.induction_with_nat_degree_le Polynomial.induction_with_natDegree_le
 
+theorem coeff_mul_add_of_le_natDegree_of_eq {df dg : ℕ} {g : R[X]}
```

Comment:
This seems like a strange file for this lemma. Is there really no easy proof without using `Polynomial.induction_with_natDegree_le`?

Reply:
I tried a little proving this result without this induction.  I also tried moving this induction much earlier in the import hierarchy.  Both these attempts have been unsuccessful.

I would be happy to see progress in either direction, though!  :smile: